packages/jellyfish-api-core/__tests__/category/loan/placeAuctionBid.test.ts

```diff
@@ -0,0 +1,643 @@
+      : undefined
+    const bobTSLAAmtAfter = bobTSLAAccAfter !== undefined ? Number(bobTSLAAccAfter.split('@')[0]) : 0
+
+    expect(bobTSLAAmtBefore - bobTSLAAmtAfter).toStrictEqual(526) // 545.45454546 - 19.45454546 = 526
+
+    const aliceColAccBefore = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccBefore).toStrictEqual(['30000.00000000@DFI', '29999.00000000@BTC', '10000.00000000@TSLA'])
+
+    // test second round placeAuctionBid
+    await alice.rpc.loan.placeAuctionBid({
+      vaultId: bobVaultId,
+      index: 0,
+      from: aliceColAddr,
+      amount: '535@TSLA' // (526 * 1%) + 525 = 531.26
+    })
+    await alice.generate(1)
+    await tGroup.waitForSync()
+
+    // end the auction and alice win the bid
+    await bob.generate(36)
+
+    const auctionsAfter = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual([
+      {
+        vaultId: bobVaultId,
+        state: 'inLiquidation',
+        loanSchemeId: 'scheme',
+        ownerAddress: bobVaultAddr,
+        liquidationPenalty: 5,
+        liquidationHeight: 205,
+        batchCount: 1,
+        batches: [
+          {
+            index: 0,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '500.00428078@TSLA' // 535 / 1.05, https://github.com/DeFiCh/pinkpaper/tree/main/loan#collateral-auction
+          }
+        ]
+      }
+    ])
+    /**
+     * The pattern is tested by several last bid amount
+     *
+     * | last bid     | loan         | recovered    | last bid vs recovered |
+     * |--------------|--------------|--------------|-----------------------|
+     * | 525.00419516 | 500.00399539 | 500.00399539 | 1.05                  |
+     * | 530          | 500.00399539 | 504.9998002  | 1.049505366           |
+     * | 535          | 500.00399539 | 509.9998002  | 1.049020019           |
+     * | 550          | 500.00399539 | 524.9998002  | 1.047619446           |
+     * | 580          | 500.00399539 | 554.9998002  | 1.045045421           |
+     * | 600          | 500.00399539 | 574.9998002  | 1.043478623           |
+     */
+
+    const bobColAccEndBid = await bob.rpc.account.getAccount(bobColAddr)
+    // compare to bobColAccAfter ['8900.00000000@DFI', '20.45454546@TSLA']
+    // bob claims back his funds
+    expect(bobColAccEndBid).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+    const aliceColAccEndBid = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccEndBid).toStrictEqual(['35000.00000000@DFI', '29999.50000000@BTC', '9465.00000000@TSLA'])
+
+    // ensure interest is freeze in auction
+    await alice.generate(10)
+    const auctionsAfter1 = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual(auctionsAfter1)
+
+    await bob.generate(40)
+
+    // should not allow place bid
+    const txid1 = await alice.rpc.loan.placeAuctionBid({
+      vaultId: bobVaultId,
+      index: 0,
```

Comment:
Okay, you are right. It can be bid again because the vault has two indexes
Once index 0 is won by someone, index = 1 will become index = 0. And this is why, the second bidding on index 0 is still allowed here.

Reply:
I think we probably want to inform the C++ side that we should not allow element INDEX to be reorganised. Imagine someone is bidding for index 1 and suddenly he is required to bid for 0. A bit confusing here.

Reply:
that's the design... i think better to hold this discussion internally

Reply:
You see, assume that wallet is pointing to index 3 and let the user bid. 
Suddenly 2 indexes won and index 3 become 1. Wallet does not know what is the new number for 3